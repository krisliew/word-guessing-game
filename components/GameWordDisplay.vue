<template>
  <section class="word-display">
    <span
      v-for="(c, index) of splitWord"
      :key="c + index"
      :class="{ character: isCharacter(c) }"
      v-html="htmlForCharacter(c)"
    ></span>
  </section>
</template>

<script>
/* eslint-disable */
import { mapGetters } from "vuex";
export default {
  name: "GameWordDisplay",
  computed: {
    ...mapGetters("game", ["currentGame"]),
    splitWord() {
      return this.currentGame ? this.currentGame.word.split("") : [];
    }
  },
  methods: {
    isCharacter(c) {
      return c !== " ";
    },
    htmlForCharacter(c) {
      if (c === " " || c === "_") {
        return "&nbsp;";
      }

      return c;
    }
  }
};
</script>

<style scoped>
.word-display {
  display: flex;
  justify-content: center;
  margin-bottom: 30px;
}

.word-display > span {
  font-size: 3rem;
  width: 1em;
  margin: 0 0.1em;
}

.character {
  border-bottom: 2px solid black;
  text-align: center;
}
</style>
