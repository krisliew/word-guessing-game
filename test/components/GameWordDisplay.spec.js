import GameWordDisplay from "@/components/GameWordDisplay.vue";
import { makeVueMock } from "@/test/utils";

describe("GameWordDisplay", () => {
  test("is vue instance", () => {
    const { wrapper } = makeVueMock(GameWordDisplay);

    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  /**
   * Snapshots
   */

  test("without word", () => {
    const { wrapper } = makeVueMock(GameWordDisplay);
    expect(wrapper.element).toMatchSnapshot();
  });

  test("with word", () => {
    const { wrapper, store } = makeVueMock(GameWordDisplay);

    store.commit("game/setCurrentGame", {
      id: "123",
      word: "_o__r B__r",
      characterGuesses: ["o", "r", "b"],
      wordGuesses: [],
      won: false
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
