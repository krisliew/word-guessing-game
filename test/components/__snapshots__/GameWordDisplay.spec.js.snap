// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameWordDisplay with word 1`] = `
<section
  class="word-display"
>
  <span
    class="character"
  >
     
  </span>
  <span
    class="character"
  >
    o
  </span>
  <span
    class="character"
  >
     
  </span>
  <span
    class="character"
  >
     
  </span>
  <span
    class="character"
  >
    r
  </span>
  <span
    class=""
  >
     
  </span>
  <span
    class="character"
  >
    B
  </span>
  <span
    class="character"
  >
     
  </span>
  <span
    class="character"
  >
     
  </span>
  <span
    class="character"
  >
    r
  </span>
</section>
`;

exports[`GameWordDisplay without word 1`] = `
<section
  class="word-display"
/>
`;
